### PR TITLE
Refactorings/transport

### PR DESCRIPTION
**Pull request by qiangdavidliu**

### CODE REVIEW

File: eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * @author David Liu
+ */
+public interface EurekaTransportConfig {
+
+    /**
+     * @return the reconnect inverval to use for sessioned clients
+     */
+    int getSessionedClientReconnectIntervalSeconds();
```
Comment (tbak) [inline eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java]: Lets always use milliseconds. Using seconds is bad for unit tests.

File: eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * Indicates how often(in seconds) to poll for changes to the bootstrap eureka server urls
+     *
+     * @return the interval to poll for bootstrap eureka server url changes (e.g. if stored in dns)
+     */
+    int getBootstrapResolverRefreshIntervalSeconds();
```
Comment (tbak) [inline eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java]: It is never referenced in the code.

File: eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * @return the interval to poll for the async resolver.
+     */
+    int getAsyncResolverRefreshIntervalMs();
+
+    /**
+     * @return the async refresh timeout threshold in ms.
+     */
+    int getAsyncResolverWarmUpTimeoutMs();
```
Comment (tbak) [inline eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java]: It is never referenced in the code.

File: eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * By default, the transport uses the same (bootstrap) resolver for queries.
+     *
+     * Set this property to false to use an indirect resolver to resolve query targets
+     * via {@link #getReadClusterVip()}. This indirect resolver may or may not return the same
+     * targets as the bootstrap servers depending on how servers are setup.
+     *
+     * @return true by default.
+     */
+    boolean useBootstrapResolverForQuery();
```
Comment (tbak) [inline eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java]: It is never referenced in the code.

File: eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java (hunk 1 above)
Comment (qiangdavidliu) [inline eureka-client/src/main/java/com/netflix/discovery/shared/transport/EurekaTransportConfig.java]: These were due to a bad diff.

**Pull request merged**